crates/l2/docs/sequencer.md

```diff
@@ -64,22 +64,23 @@ The following environment variables are available to configure the Proposer cons
 
 - Under the `[watcher]` section:
 
-  - `bridge_address`: Address of the bridge contract on L1.
-  - `check_interval_ms`: Interval in milliseconds to check for new events.
-  - `max_block_step`: Maximum number of blocks to look for when checking for new events.
-  - `l2_proposer_private_key`: Private key of the L2 proposer.
+  - `bridge_address`: Address of the bridge contract on L1. This address is used to retrieve logs emitted by deposit events.
+  - `check_interval_ms`: Interval in milliseconds to check for new events. If no new events or messages are detected, it does nothing.
+  - `max_block_step`: Defines the maximum range of blocks to scan for new events during each polling cycle. Specifically, events are queried from last_block_fetched up to last_block_fetched + max_block_step. If the chain hasn’t progressed that far yet, the scan will end at the current latest block instead. This ensures we only query blocks that actually exist.
+  - `l2_proposer_private_key`: Private key of the L2 proposer. 
 
 - Under the `[proposer]` section:
 
-  - `interval_ms`: Interval in milliseconds to produce new blocks for the proposer.
+  - `interval_ms`: Interval in milliseconds at which the proposer wakes up to produce a new block.
```

Comment:
Interval in milliseconds at which the proposer builds a new block out of the current transactions on the mempool